out_writeapi_test.go

```diff
@@ -38,3 +44,143 @@ func TestFLBPluginRegister(t *testing.T) {
 	assert.Equal(t, "writeapi", currplugin.name)
 
 }
+
+type StreamChecks struct {
+	calledGetContext     bool
+	calledcheckResponses bool
+	createDecoder        bool
+	gotRecord            bool
+	calledparseMap       bool
+	calledJTB            bool
+	appendRows           bool
+	appendQueue          bool
+}
+
+type MockManagedStream struct {
+	managedstream  *managedwriter.ManagedStream
+	AppendRowsFunc func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	StreamName     func() string
+	CloseFunc      func() error
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	var checks StreamChecks
+	var setID interface{}
+	var id int
+
+	mockMS := &MockManagedStream{
+		AppendRowsFunc: func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+			checks.appendRows = true
+			return nil, nil
+		},
+	}
+
+	originalFunc := getWriter
+	getWriter = func(client *managedwriter.Client, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = originalFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		return ""
```

Comment:
I think you will have to provide a valid string for projectid, datasetid and tableid. Otherwise the plugin will fail the call to Init().

Reply:
I can add a dummy projectID, but we don't need datasetID or tableID for anything other than NewMangedStream, which I mocked to not need these. Would you still want me to add a dummy datasetID and tableID?

Reply:
If it works then that's fine. I was thinking it might not work after integrating with raahi's pending changes but that may not be the case.

Reply:
In that case I'm just going to leave it for now!

Reply:
Done.